File: app/src/main/java/com/romellfudi/ussd/di/AppComponent.java
package com.romellfudi.ussd.di;


import com.romellfudi.ussd.sample.MainActivity;
import com.romellfudi.ussd.sample.MainFragment;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 2020-04-20
 */

@Singleton
@Component(modules = {USSDModule.class})
public interface AppComponent {

    void inject(MainFragment fragment);
    void inject(MainActivity activity);
}

File: app-kotlin/src/main/java/com/romellfudi/ussd/di/AppComponent.kt
package com.romellfudi.ussd.di

import android.content.Context
import com.romellfudi.ussd.sample.UIComponent
import dagger.BindsInstance
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [USSDModule::class, AppSubcomponents::class])
interface AppComponent {

    @Component.Factory
    interface Factory {
        fun create(@BindsInstance context: Context): AppComponent
    }

    fun uiComponent(): UIComponent.Factory
}
